fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java

```diff
@@ -150,7 +150,27 @@
     // If low cardinality string column with global dict, for some string functions,
     // we could evaluate the function only with the dict content, not all string column data.
     public final ImmutableSet<String> couldApplyDictOptimizationFunctions = ImmutableSet.of(
-            "like", "substr", "substring", "upper", "lower");
+           "append_trailing_char_if_absent",
+           "concat",
+           "concat_ws",
+           "hex",
+           "left",
+           "like", 
+           "lower",
+           "lpad",
+           "ltrim",
+           "regexp_extract",
+           "regexp_replace",
+           "repeat",
+           "reverse",
+           "right",
+           "rpad",
+           "rtrim",
+           "split_part",
+           "substr", 
+           "substring", 
+           "trim",
+           "upper");
```

Comment:
start_with?
end_with?
It's looks can be work for most of string functions, you can be find these in vectorized_functions.py

Reply:
now only expr (string,...)->string was support